### DISCUSSION TITLE
No Backup Being Created (even after config.json was updated)

### DISCUSSION THREAD

**Discussion by Vortain** [Pryaxis/TShock#1703] [Q&A]
I've got my server up and running successfully, but I can't seem to get TShock to backup the world on regular intervals.  I updated the config.json file within the TShock folder, and set the interval to 30 minutes and 14400 for the "BackupKeepFor".  I validated the json file and it's fine.

There is no backup folder and obviously no backup worlds being created from what I can tell.  I tried changing the port # to 7778 just to see if that would make a difference when restarting/reloading the server, and the default is still 7777.  I know that tshock is at least touching the config file as, when I have the config file in the editor and restart the server, the editor tells me it's been modified and asks if I want to reload it.

I'm worried we could lose data if the worlds get corrupted or a hacker finds their way in.  Any help is appreciated!

**Comment by moisterrific** [comment]
Not sure if backups work since I’ve never tried using it, but I can test tomorrow to see if it’s a bug inherent in TShock. 

As for server port, I think you have to specify the port when starting the server, unless you start it with parameters. I couldn’t get the config defined port to work either since pressing Enter defaults to 7777.

**Comment by QuiCM** [comment]
The precedence order on the port is a little weird. Best bet is to set it via commandline as that should be top priority

**Comment by bippity** [comment]
I decided to check out my own servers' backups and noticed that something's off. I've configured two servers to have 2-hour backup intervals and the backups will be retained for 72 hours (3 days)... 

On **Server 1**, here are the things I've noticed:
- The backups I see are between [04/23 - 04/24] (today is 04/27) so there has not been any backups for the last 3-4 days. 
- There are inconsistent time gaps between the backups which do not follow the 2-hour backup interval (11 hrs, 7 hrs, 3 hrs, and 23 hrs)
    
On **Server 2**, here are the things I've noticed:
- The backups folder was empty despite having the same config time as Server 1 (`"BackupInterval": 120, "BackupKeepFor": 4320`)
- The logs show that backups were being saved into `tshock/backups` but the folder is empty

I might need a refresher on how backups work again but this doesn't seem like normal behavior to me. 
Side note: Both servers have been running continuously since August 2019 and I haven't touched them since. Not sure if that would cause any issues but, I've rebooted them to see if the backups will start again. 

```TL;DR: One server hasn't made any backups for the past 3 days and the other server states that backups were made but, the folder is empty```

**Reply by hakusaro** [reply]
Have you had active players on the days that you haven't seen backups?

**Reply by bippity** [reply]
**Server 1** is actually a private server, some Russian dude managed to find the server but disconnected shortly awhile back.
**Server 2** is a public server where players join and disconnect pretty often.

After rebooting the servers, Server 2 seems to be working normally now. I see multiple backups with the correct 2-hour intervals. 
Server 1 still hasn't made any recent backup but I'm assuming that's because no player has joined it recently?

**Reply by hakusaro** [reply]
So, no active players? Ergo, nothing has changed on the world?

**Reply by bippity** [reply]
Yeap, I just triggered a backup on Server 1 by joining it so it's also working normally now.

I guess the only curious part is why Server 2 had an empty backup folder despite saying it created backups.

**Reply by bippity** [reply]
It's been about 9 days since I've rebooted the servers. 
**Server 1** appears to still work correctly

But **Server 2** appears to be having the same issues as before...
* Log states that a backup has been made today (5/6) but the latest backups in the folder are 2 days old (5/4)
* The backup intervals are still happening at the correct times (2-hour intervals)

I'll try a vanilla installation on my local machine to see if this is just an isolated issue and not directly caused by tshock.

**Reply by hakusaro** [reply]
Did you try using `-forceupdate`?

**Comment by QuiCM** [comment]
May be worth converting this into an issue as from the above accounts, backups are not working as expected.

**Reply by hakusaro** [reply]
Can't actually convert from discussion to issue, but I wanna see if bippers is using `-forceupdate` or not

**Reply by Kojirremer** [reply]
> Can't actually convert from discussion to issue, but I wanna see if bippers is using `-forceupdate` or not

Not directly related to this issue, but I'll just say that -forceupdate does not seem to work. I've tried it a few times to try growing things and the server still goes into hibernation when no players are active (world "stops").
